docs/latest/design/egctl.md

```diff
@@ -0,0 +1,67 @@
+# Introduce egctl
+
+## Motivation
+
+EG should provide a command line tool with following capabilities:
+
+- One-step installation/uninstallation experience
+- Collect configuration from envoy proxy and gateway
+- Update admin options such as changing log level
+- Analyse system configuration to diagnose any issues in envoy gateway
+
+This tool is named `egctl`.
+
+## Syntax
+
+Use the following syntax to run `egctl` commands from your terminal window:
+
+```console
+egctl [command] [subcommand] [name] [flags]
+```
+
+where `command`, `name`, and `flags` are:
+
+* `command`: Specifies the operation that you want to perform on one or more resources,
+  for example `install`, `uninstall`, `config`, `version`.
+
```

Comment:
can you add a explanation for `subcommand` or replace it with another term such as `entity` or `resource` : Specifies the entity the operation is being performed on such as `envoy-proxy` or `envoy-gateway`

Reply:
Not a blocker but I prefer `control-plane` and `data-plane` over `envoy-gateway` and `envoy-proxy`.

Reply:
@danehans using actual names (`envoy-gateway`, `envoy-proxy`) allows us to fetch info from specific resources and can be extended to other resources such as `envoy-ratelimit-service` and other auxiliary control planes and data planes